pkg/gofr/datasource/solr/solr.go

```diff
@@ -0,0 +1,215 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+)
+
+type Config struct {
+	Host string
+	Port string
+}
+
+type Client struct {
+	url string
+
+	logger  Logger
+	metrics Metrics
+}
+
+// New initializes Solr driver with the provided configuration.
+// The Connect method must be called to establish a connection to Solr.
+// Usage:
+// client := New(config)
+// client.UseLogger(loggerInstance)
+// client.UseMetrics(metricsInstance)
+// client.Connect()
+func New(conf Config) *Client {
+	s := &Client{}
+	s.url = "http://" + conf.Host + ":" + conf.Port + "/solr/"
```

Comment:
Is solr always in http?, so not in https

Reply:
I believe it can be supported with https as well.. but since these calls will be within the same cluster, hence using the `http`

Reply:
shouldn't that be configurable then?